src/test/java/com/redhat/pantheon/servlet/DefaultQueryServletTest.java

```diff
@@ -0,0 +1,203 @@
+        assertEquals(10L, jsonNode.get("size").asLong());
+        assertEquals(10L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
+        Iterator<JsonNode> resultIterator = jsonNode.get("results").iterator();
+        // verify the order is inversed
+        for (int i = 0; i < 10; i++) {
+            assertEquals("node-" + (9-i), resultIterator.next().get("name").asText());
+        }
+    }
+
+    @Test
+    @DisplayName("Test a filtered query")
+    public void testFilteredQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
+        slingContext.request().getParameterMap().put("where", new String[]{"number < 5"});
+
+        // When
+        servlet.doGet(slingContext.request(), slingContext.response());
+
+        // Then
+        JsonNode jsonNode = new ObjectMapper().readTree(slingContext.response().getOutputAsString());
+        assertEquals(5L, jsonNode.get("size").asLong());
+        assertEquals(5L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
+        Iterator<JsonNode> resultIterator = jsonNode.get("results").iterator();
+        // verify the right nodes are included
+        resultIterator.forEachRemaining(n -> assertTrue(n.get("number").asLong() < 5));
+    }
+
+    @Test
+    @DisplayName("Test paged Query")
+    public void testPagedQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
+        slingContext.request().getParameterMap().put("limit", new String[]{"5"});
+
+        // When
+        servlet.doGet(slingContext.request(), slingContext.response());
+
+        // Then
+        JsonNode jsonNode = new ObjectMapper().readTree(slingContext.response().getOutputAsString());
+        assertEquals(5L, jsonNode.get("size").asLong());
+        assertEquals(5L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
+        for (int i = 0; i < 5; i++) {
+            assertTrue(slingContext.response().getOutputAsString().contains("node-" + i));
```

Comment:
I would change this to just verify that the output contains 5 nodes. If you want to be specific about which 5 nodes it contains, then I would specify a sort order as well.

Reply:
:+1: removed the check